crates/dbsp/src/algebra/mod.rs

```diff
@@ -29,6 +31,95 @@ use std::{
     rc::Rc,
 };
 
+/// Trait for types where the minimum value is known
+trait MinValue {
+    fn min_value() -> Self;
+}
+
+/// Trait for types where the maximum value is known
+trait MaxValue {
+    fn max_value() -> Self;
+}
+
+/// Trait for integer types returning the first
+/// value representable in this type which is
+/// too large for the previous narrower type
+pub trait FirstLargeValue {
+    fn large() -> Self;
+}
+
+/// Trait for primitive integers that are unsigned
+pub trait IsUnsigned: PrimInt + FirstLargeValue + HasZero + Debug + Display {}
```

Comment:
It's odd that `IsUnsigned` is derived from `PrimInt`. Maybe `UnsignedPrimInt` is a better name for it?